client/src/app/containers/Alert/index.tsx

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+import { useSelector } from 'react-redux';
+
+import { RootState } from '../../../store/store';
+
+const Alert: React.FunctionComponent = () => {
+    const alerts = useSelector((state: RootState) => state.alerts);
```

Comment:
:mag: why do components even get to ask for state? Shouldn't we always be giving a component all the relevant state, and then it just renders that for us? Why does it get to have access to `RootState`?

Reply:
So, we could totally pass state down through props. This is canonically called prop drilling. The problem is, if I need to pass some value all the way from the top level (`App`) to a deeply nested child component, I have to pass this data through all the layers in between. Want to change the type of this value? You have to edit the type of the prop on every component it gets passed through. Want to pass more data? You have to define it as a prop on every level.

The whole point of redux is that state is accessible at any level in the component hierarchy. If I want to grab another value from state here, I don't have to change several other components to do so.